tests/integration_tests/integration_settings.py

```diff
@@ -0,0 +1,95 @@
+# This file is part of cloud-init. See LICENSE file for license information.
+import os
+
+##################################################################
+# LAUNCH SETTINGS
+##################################################################
+
+# Keep instance (mostly for debugging) when test is finished
+KEEP_INSTANCE = False
+
+# One of:
+#  lxd_container
+#  ec2
+#  gce
+#  oci
+PLATFORM = 'lxd_container'
+
+# The cloud-specific instance type to run. E.g., a1.medium on AWS
+# If the pycloudlib instance provides a default, this can be left None
+INSTANCE_TYPE = None
+
+# Determines the base image to use or generate new images from.
+# Can be the name of the OS if running a stock image,
+# otherwise the id of the image being used if using a custom image
+OS_IMAGE = 'focal'
+
+# Populate if you want to use a pre-launched instance instead of
+# creating a new one. The exact contents will be platform dependent
+EXISTING_INSTANCE_ID = None
+
+##################################################################
+# IMAGE GENERATION SETTINGS
+##################################################################
+
+# Depending on where we are in the development / test / SRU cycle, we'll want
+# different methods of getting the source code to our SUT. Because of
+# this there are a number of different ways to initialize
+# the target environment.
+
+# Can be any of the following:
+# NONE
+#   Don't modify the target environment at all. This will run
+#   cloud-init with whatever code was baked into the image
+# IN_PLACE
+#   LXD/VM only. Mount the source code as-is directly into
+#   the container to override the pre-existing cloud-init code
+# PROPOSED
+#   Install from the proposed repo
+# PPA
+#   Install from a PPA
+# <file path>
+#   A path to a valid package to be uploaded and installed
+# <image id>
+#   A pre-existing (platform dependent) image id
+IMAGE_SOURCE = 'NONE'
+
+##################################################################
+# GCE SPECIFIC SETTINGS
+##################################################################
+# Required for GCE
+GCE_PROJECT = None
+
+# You probably want to override these
+GCE_REGION = 'us-central1'
+GCE_ZONE = 'a'
+
+##################################################################
+# OCI SPECIFIC SETTINGS
+##################################################################
+# Compartment-id found at
+# https://console.us-phoenix-1.oraclecloud.com/a/identity/compartments
+# Required for Oracle
+OCI_COMPARTMENT_ID = None
+
+##################################################################
+# USER SETTINGS OVERRIDES
+##################################################################
+# Bring in any user-file defined settings
+try:
+    from tests.integration_tests.user_settings import *
+except:  # noqa
+    pass
+
+##################################################################
+# ENVIRONMENT SETTINGS OVERRIDES
```

Comment:
At some point in might be worth providing a basic argparse for these switches too (as setting env variables CLOUD_INIT_* for any param is bound to get verbose fairly quickly.

Reply:
I'd actually rather not for a few reasons. 

For customizing things on your individual machine, you can use the file I added to the gitignore: https://github.com/canonical/cloud-init/pull/581#diff-a084b794bc0759e7a6b77810e01874f2R32
Anything in integration_settings.py will be overridden by anything in user_settings.py. Anything in either of those two  will be overridden by anything in an env var.

If we're talking about scripting or running on CI, I think env vars are easier. You can specify them on different lines or source an entire file of them, rather than relying on a single line of a million variable substitutions.

Additionally, since this is being run via pytest, you can't really use argparse directly. You have to hook into pytest. The `pytest -h` output is already fairly verbose, and adding more flags won't exactly make it more user friendly. Also, I wouldn't want them to be displayed when we're running our unit tests, and I'm not sure how easy it would be to customize that.